uswds/uswds
# USWDS - Select: Multiple select overflow

# Summary

**Added ellipses to multiple select overflow.** This provides a clear indication to users that there is text being cut off by the select width.

## Breaking change

This is not a breaking change.

## Related issue

Closes #5047 

## Related pull requests

[Site Changelog PR](https://github.com/uswds/uswds-site/pull/2080)

## Preview link

[Multiple select →](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/cm-multiple-select-overflow/iframe.html?args=&id=components-form-inputs-select--multiple&viewMode=story)

## Problem statement

Multiple select cuts off long options on Safari & Firefox due to browser styles changing the text color to white, and additional padding leftover from the single select's dropdown icon

## Solution


Unfortunately, there is no way to wrap option text within a select element at this time. Instead, the component should convey that there is additional text not visible to the user. This will provide a better user experience for all multiple select users.

This still isn't ideal for users and developers should focus on allowing options to be completely visible within their selects by either increasing the select element width or using short, concise options.

## Testing and review

1. Visit multiple select story in Safari or Firefox
2. View the longer `Option C` 
3. Confirm ellipses are visible
4. Select option C
5. Confirm no information loss due to padding or text color change

## Screenshots

### Before

**Chrome:**
![image](https://user-images.githubusercontent.com/25211650/235754132-b9f8ebe3-e3fc-442b-885c-ae8fd45c278f.png)


**Firefox**
![image](https://user-images.githubusercontent.com/25211650/235754039-8219a247-6763-4644-87d8-b5ca63252481.png)

**Safari**
![image](https://github.com/uswds/uswds/assets/25211650/b2235510-e1aa-4205-bdf6-ad901fea8b89)


### After

**Chrome:**
![image](https://github.com/uswds/uswds/assets/25211650/3501c009-368b-47cc-9e24-ee4b5f46f7ce)

**Firefox**
![image](https://github.com/uswds/uswds/assets/25211650/4ecd837d-eea5-4848-9ac1-10d5a5a8dfcd)

**Safari**
![image](https://github.com/uswds/uswds/assets/25211650/cba851db-051a-4466-90ea-e9a5c2601a4b)
